modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -391,10 +391,63 @@ void ONNXImporter::populateNet(Net dstNet)
                     CV_Error(Error::StsNotImplemented, "Unsupported mode of ReduceMean operation.");
 
                 MatShape inpShape = outShapes[node_proto.input(0)];
+                DictValue axes = layerParams.get("axes");
+                if (inpShape.size() == 3 && axes.size() <= 2)
+                {
+                    int axis = axes.get<int>(0);
+                    CV_CheckNE(axis, 0, "");
+                    outShapes[layerParams.name] = inpShape;
+                    outShapes[layerParams.name][axis] = 1;
+
+                    LayerParams reshapeLp;
+                    std::string reshapeName = layerParams.name + "/reshape";
+                    CV_Assert(layer_id.find(reshapeName) == layer_id.end());
+                    reshapeLp.set("axis", 0);
+                    reshapeLp.set("num_axes", 1);
+                    int newShape[] = {1, -1};
+                    reshapeLp.set("dim", DictValue::arrayInt(&newShape[0], 2));
+                    int reshapeId = dstNet.addLayer(reshapeName, "Reshape", reshapeLp);
+                    layerId = layer_id.find(node_proto.input(0));
+                    CV_Assert(layerId != layer_id.end());
+                    dstNet.connect(layerId->second.layerId, layerId->second.outputId, reshapeId, 0);
+                    layer_id.insert(std::make_pair(reshapeName, LayerInfo(reshapeId, 0)));
```

Comment:
Please try to replace to `addLayer` from https://cgithub.com/opencv/opencv/pull/16817